.github/workflows/changeset-enforce.yml

```diff
@@ -0,0 +1,43 @@
+name: No patch level changesets
+on:
+  pull_request
+
+jobs:
+  check_changesets:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout repository (feature branch)
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+          fetch-depth: 1
+
+      - name: Checkout repository (develop branch)
+        uses: actions/checkout@v4
+        with:
+          ref: develop
+          fetch-depth: 1
+
+      - uses: pnpm/action-setup@v4
+        with:
+          version: 9.11.0
+
+      - name: Install changesets
+        # run: pnpm install -w @changesets/cli --no-frozen-lockfile
+        run: pnpm add -g @changesets/cli
```

Comment:
i would recommend to pin this to avoid issues with latest versions of cli 
`pnpm add -g @changesets/cli@2.27.7`